ProjectCoimbra.UWP/Project.Coimbra/Pages/GamePage.xaml.cs

```diff
@@ -2,217 +2,343 @@
-            {
-                this.timer.Tick += this.Timer_Tick;
-                this.timer.Interval = TimeSpan.FromSeconds(1);
-                this.timer.Start();
-            }
-            else
-            {
-                this.InputControl.Pitches = UserData.PitchMap.Pitches;
-                this.midiEngine.Start();
-            }
-        }
-
-        private static int? ConvertToLane(NoteEvent currentNoteOnEvent)
-        {
-            var noteName = currentNoteOnEvent.GetNoteName().ToString();
-            return Array.Find(UserData.PitchMap.Pitches, pitch => pitch.NoteNames.Any(
-                    (pitchNoteName) => string.Equals(pitchNoteName, noteName, StringComparison.Ordinal)))?.Index;
-        }
+	using Coimbra.Controls;
+	using Coimbra.DryWetMidiIntegration;
+	using Coimbra.Midi;
+	using Coimbra.Model;
+	using Melanchall.DryWetMidi.Core;
+	using Melanchall.DryWetMidi.Interaction;
+	using Microsoft.Toolkit.Uwp.Input.GazeInteraction;
+	using System;
+	using System.Collections.Concurrent;
+	using System.Collections.Generic;
+	using System.Globalization;
+	using System.Linq;
+	using System.Runtime.InteropServices;
+	using System.Threading.Tasks;
+	using Windows.ApplicationModel.Core;
+	using Windows.UI.Core;
+	using Windows.UI.Xaml;
+	using Windows.UI.Xaml.Controls;
+	using Windows.UI.Xaml.Media.Animation;
+
+	/// <summary>
+	/// A class encapsulating the logic of the game page of the app.
+	/// </summary>
+	public sealed partial class GamePage : Page
+	{
+		private readonly MidiEngine midiEngine = MidiEngine.Instance;
+
+		private readonly ConcurrentDictionary<Guid, RenderedMarkablePlaybackEvent> notesOnScreen =
+			new ConcurrentDictionary<Guid, RenderedMarkablePlaybackEvent>();
+
+		private readonly DispatcherTimer timer = new DispatcherTimer();
+		private readonly TimeSpan noteDuration = TimeSpan.FromSeconds(5);
+
+		private List<long> noteTimes;
+		private int lastPlayedNoteTimeIndex = 0;
+		private int dotCounter = 0;
+		private long previousTimeToNote = 0;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="GamePage"/> class.
+		/// </summary>
+		public GamePage()
+		{
+			if (UserData.ActiveDuration == UserData.Duration.UnknownDuration)
+			{
+				UserData.ActiveDuration = UserData.Duration.MediumDuration;
+			}
+
+			this.InitializeComponent();
+			this.midiEngine.PlaybackFinished += this.MidiEngine_PlaybackFinished;
+			this.midiEngine.RenderCurrentNotesAsyncEvent += this.MidiEngine_RenderCurrentNotesEventAsync;
+			this.InputControl.LaneButtonClicked += this.InputControl_LaneButtonClicked;
+			this.InputControl.EyeGazeInteracted += this.InputControl_EyeGazeInteracted;
+			this.InputControl.SongTitle = this.midiEngine.TrackDisplayName;
+			this.InputControl.SecondsDuringWhichNoteIsActive = (int)UserData.ActiveDuration;
+
+			this.midiEngine.Initialize();
+			noteTimes = this.midiEngine.RetrieveNoteTimesForInstrument(this.midiEngine.SelectedTrack);
+
+			this.lastPlayedNoteTimeIndex = 0;
```

Comment:
I don't believe you need to set these, as they're already set above during class initialisation.

Also, they don't even need to be set there as integer values always default to 0 in .NET.